### PR TITLE
postgresql_user.py - Documentation

### PR DESCRIPTION
**Pull request by vrms**
added info about using SCRAM-SHA-256 (& not only md5) hashed passwords.

### CODE REVIEW

File: plugins/modules/postgresql_user.py [hunk 1]
```diff
@@ -213,12 +213,15 @@
     name: django
     comment: This is a test user
 
-# Connect to default database, create rails user, set its password (MD5-hashed),
+# Connect to default database, create rails user, set its password (MD5- or sha256-hashed),
 # and grant privilege to create other databases and demote rails from super user status if user exists
+# for SCRAM-SHA-256 you can create the hash by creating a dummy user with the same password & copying 
+# the hash from the corresponding pg_authid entry.
```
Comment (Andersson007) [inline plugins/modules/postgresql_user.py:219]:
```suggestion
# for SCRAM-SHA-256 you can create the hash by creating a dummy user with the same password & copying 
# the hash from the corresponding pg_authid entry.
```

File: plugins/modules/postgresql_user.py [hunk 2]
```diff
@@ -213,12 +213,15 @@
     name: django
     comment: This is a test user
 
-# Connect to default database, create rails user, set its password (MD5-hashed),
+# Connect to default database, create rails user, set its password (MD5- or sha256-hashed),
 # and grant privilege to create other databases and demote rails from super user status if user exists
+# for SCRAM-SHA-256 you can create the hash by creating a dummy user with the same password & copying
```
Comment (Andersson007) [inline plugins/modules/postgresql_user.py:218]:
```suggestion
# for SCRAM-SHA-256 you can create the hash by creating a dummy user with the same password & copying
```

File: plugins/modules/postgresql_user.py [hunk 3]
```diff
@@ -213,12 +213,15 @@
...
+# the hash from the corresponding pg_authid entry.
 - name: Create rails user, set MD5-hashed password, grant privs
   community.postgresql.postgresql_user:
     name: rails
     password: md59543f1d82624df2b31672ec0f7050460
+    # password: SCRAM-SHA-256$4096:zFuajwIVdli9mK=NJkcv1Q++$JC4gWIrEHmF6sqRbEiZw5FFW45HUPrpVzNdoM72o730+;fqA4vLN3mCZGbhcbQyvNYY7anCrUTsem1eCh/4YA94=
     role_attr_flags: CREATEDB,NOSUPERUSER
```
Comment (Andersson007) [inline plugins/modules/postgresql_user.py:225]:
```suggestion
    role_attr_flags: CREATEDB,NOSUPERUSER
  # When using sha256-hashed password:
  #environment:
  #  PGOPTIONS: "-c password_encryption=scram-sha-256"
```

**Review by Andersson007 — APPROVED** [review]
LGTM

**Review by jchancojr — APPROVED** [review]

File: plugins/modules/postgresql_user.py [hunk 4]
```diff
@@ -213,13 +213,19 @@
     name: django
     comment: This is a test user
 
-# Connect to default database, create rails user, set its password (MD5-hashed),
+# Connect to default database, create rails user, set its password (MD5- or sha256-hashed),
```
Comment (hunleyd) [inline plugins/modules/postgresql_user.py:216]:
```suggestion
# Connect to default database, create rails user, set its password (MD5- or SHA256-hashed),
```

File: plugins/modules/postgresql_user.py [hunk 5]
```diff
@@ -213,13 +213,19 @@
...
 # and grant privilege to create other databases and demote rails from super user status if user exists
+# for SCRAM-SHA-256 you can create the hash by creating a dummy user with the same password & copying
```
Comment (hunleyd) [inline plugins/modules/postgresql_user.py:218]: while this is true, it seems wastful. there's a python script [here](https://gist.github.com/jkatz/e0a1f52f66fa03b732945f6eb94d9c21) by a PG author that can be used instead

**Review by hunleyd — CHANGES_REQUESTED** [review]

File: plugins/modules/postgresql_user.py (hunk 5 above)
Comment (Andersson007) [reply plugins/modules/postgresql_user.py:218]: maybe we could just remove this statement?

Comment (Andersson007) [reply plugins/modules/postgresql_user.py:218]: @vrms thoughts ^

File: plugins/modules/postgresql_user.py [hunk 6]
```diff
@@ -213,13 +213,19 @@
     name: django
     comment: This is a test user
 
-# Connect to default database, create rails user, set its password (MD5-hashed),
+# Connect to default database, create rails user, set its password (MD5- or SHA256-hashed),
 # and grant privilege to create other databases and demote rails from super user status if user exists
+# for SCRAM-SHA-256 you can create the hash by creating a dummy user with the same password & copying
```
Comment (hunleyd) [inline plugins/modules/postgresql_user.py:218]:
```suggestion
```

**Review by hunleyd — CHANGES_REQUESTED** [review]

File: plugins/modules/postgresql_user.py (hunk 5 above)
Comment (hunleyd) [reply plugins/modules/postgresql_user.py:218]: removing based on Matrix discussion

**Review by hunleyd — APPROVED** [review]

**Pull request merged**